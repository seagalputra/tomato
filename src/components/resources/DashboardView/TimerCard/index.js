import React from 'react'
import PropTypes from 'prop-types'

import Button from 'components/common/Button'

const status = {
  FOCUS: 'Focus',
  LONG_BREAK: 'Long Break',
  SHORT_BREAK: 'Short Break'
}

const TimerCard = ({ onStart, onStop, state }) => {
  return (
    <div className="flex flex-col border rounded p-6 max-w-md">
      <div className="flex justify-center">
        <h1
          className={`${
            state === 'LONG_BREAK' || state === 'SHORT_BREAK'
              ? 'text-green-600'
              : 'text-red-600'
          } font-bold text-xl`}
        >
          {status[state]}
        </h1>
      </div>

      <div className="flex justify-center mt-2">
        <p className="text-gray-800 font-bold text-5xl">25:00</p>
      </div>

      <div className="inline-flex justify-center w-auto mt-4">
        <Button className="mr-2" onClick={onStart} variant="primary">
          Start
        </Button>

        <Button onClick={onStop} variant="secondary">
          Stop
        </Button>
      </div>
    </div>
  )
}

TimerCard.propTypes = {
  state: PropTypes.string.isRequired,
  onStart: PropTypes.func,
  onStop: PropTypes.func
}

export default TimerCard
