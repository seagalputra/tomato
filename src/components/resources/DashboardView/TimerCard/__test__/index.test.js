import React from 'react'
import { shallow } from 'enzyme'
import TimerCard from '..'

describe('TimerCard', () => {
  it('renders without crashing', () => {
    const state = 'FOCUS'

    const component = shallow(<TimerCard state={state} />)
    expect(component).toMatchSnapshot()
  })
})
