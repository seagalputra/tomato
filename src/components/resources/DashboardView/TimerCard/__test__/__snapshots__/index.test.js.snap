// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimerCard renders without crashing 1`] = `
<div
  className="flex flex-col border rounded p-6 max-w-md"
>
  <div
    className="flex justify-center"
  >
    <h1
      className="text-red-600 font-bold text-xl"
    >
      Focus
    </h1>
  </div>
  <div
    className="flex justify-center mt-2"
  >
    <p
      className="text-gray-800 font-bold text-5xl"
    >
      25:00
    </p>
  </div>
  <div
    className="inline-flex justify-center w-auto mt-4"
  >
    <Button
      className="mr-2"
      isDisabled={false}
      variant="primary"
    >
      Start
    </Button>
    <Button
      isDisabled={false}
      variant="secondary"
    >
      Stop
    </Button>
  </div>
</div>
`;
